repomgr/repomgr.go

```diff
@@ -885,28 +885,30 @@ func processOp(ctx context.Context, bs blockstore.Blockstore, op *mst.DiffOp) (*
 			return nil, err
 		}
 
-		rec, err := lexutil.CborDecodeValue(blk.RawData())
```

Comment:
Semantically, I think we should (must?) validate that the record as acceptable here, but it should be lexicon-agnostic. We probably don't have a helper for that yet? What I mean is things like:

- actually decodes as CBOR at all
- meets additional DAG-CBOR requirements (eg, no tag types other than 42 "link")
- meets any Lexicon requirements, eg no weird CID types (raw-inline data CID)

Basically BGS needs to be conservative about low-level details (below app schema) and permissive about higher-level stuff (above app pschema).

In this PR, would be good to at least do CBOR decode, if we have a generic way to do that. Maybe call `CborTypeExtract`? all records should (must) have a `$type`.

Reply:
Oh, now I see that you moved the `CborDecodeValue` below